### PR TITLE
doc: update documentation for 1.4.2

### PR DESCRIPTION
**Pull request by greg-fer**
Updated documentation for 1.4.2 release.
Bumped release version number.
Added files for release notes.
Add known issues for 1.4.2.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/known_issues.rst [hunk 1]
```diff
@@ -21,7 +21,8 @@ Use the drop-down filter to see known issues for previous releases and check if
 
    <select name="versions" id="versions-select">
      <option value="all">All versions</option>
-     <option value="v1-4-1" selected>v1.4.1</option>
+     <option value="v1-4-2" selected>v1.4.2</option>
```
Comment (greg-fer) [inline doc/nrf/known_issues.rst:24]: @nordic-hani , kindly please give a look at known issue status for lwm2m_carrier. I've propagated 1.4.2 to all 1.4.1 issues, but some might've been fixed in 0.10.2.

File: doc/nrf/releases/release-notes-1.4.2.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+The following sections provide detailed lists of changes by component.
+
+nRF9160
+=======
+
+* :ref:`liblwm2m_carrier_readme` library:
+
+  * Updated to v0.10.2.
+    See the :ref:`liblwm2m_carrier_changelog` for detailed information.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-1.4.2.rst:53]: Changelog edits pending in #3538 .

File: doc/versions.json [hunk 3]
```diff
@@ -19,6 +19,12 @@
       "nrfxlib": "1.4.99",
       "zephyr": "2.4.0",
       "mcuboot": "1.6.99"
+    },
+      "1.4.2": {
+      "ncs": "1.4.2",
+      "nrfxlib": "1.4.0",
+      "zephyr": "2.4.0",
+      "mcuboot": "1.6.99"
```
Comment (greg-fer) [inline doc/versions.json:27]: @tejlmand , can you kindly confirm these versions? Thanks!

File: doc/versions.json [hunk 4]
```diff
@@ -19,6 +20,12 @@
       "nrfxlib": "1.4.99",
       "zephyr": "2.4.0",
       "mcuboot": "1.6.99"
+    },
+      "1.4.2": {
+      "ncs": "1.4.2",
+      "nrfxlib": "1.4.0",
```
Comment (ru-fu) [inline doc/versions.json:26]: nrfxlib must be (at least) 1.4.1 - that was missed in the update for 1.4.1, but is fixed on master (see https://github.com/nrfconnect/sdk-nrf/pull/3570 ).
This must also be fixed in the conf.py file for nrfxlib.

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-1.4.2.rst (hunk 2 above)
Comment (nordic-hani) [reply doc/nrf/releases/release-notes-1.4.2.rst:53]: #3538  has been merged 👍

File: doc/nrf/known_issues.rst [hunk 5]
```diff
@@ -93,12 +94,12 @@ Sending data before connecting to nRF Cloud
 Other issues
 ============
 
-.. rst-class:: v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0
+.. rst-class:: v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0
```
Comment (nordic-hani) [inline doc/nrf/known_issues.rst:97]: LTE sensor gateway known issue is fixed. Do not apply 1.4.2 label here.

File: doc/nrf/releases/release-notes-1.4.2.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+To use this release, check out the tag in the manifest repository and run ``west update``.
+See :ref:`cloning_the_repositories` for more information.
+
+Supported modem firmware
+************************
+
+This version of the |NCS| has been tested with the following modem firmware for cellular IoT applications:
+
+* mfw_nrf9160_1.2.2
```
Comment (shantha-14) [inline doc/nrf/releases/release-notes-1.4.2.rst:37]: mfw_nrf9160_1.2.3

File: doc/versions.json (hunk 4 above)
Comment (shantha-14) [reply doc/versions.json:26]: agree

**Review by shantha-14 — CHANGES_REQUESTED** [review]

**Review by jorgenmk — APPROVED** [review]

File: doc/nrf/known_issues.rst [hunk 7]
```diff
@@ -90,15 +91,26 @@ Sending data before connecting to nRF Cloud
   The :ref:`asset_tracker` sample does not wait for connection to nRF Cloud before trying to send data.
   This causes the sample to crash if the user toggles one of the switches before the board is connected to the cloud.
 
+.. rst-class:: v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0
+
+Missing support for FOTA on nRF Cloud
+  The :ref:`asset_tracker` sample does not support the nRFCloud FOTA_v2 protocol.
```
Comment (ru-fu) [inline doc/nrf/known_issues.rst:97]: Two different spellings for nRF Cloud here - which should really be nRF Connect for Cloud (but I don't think we've updated that everywhere).

File: doc/versions.json (hunk 4 above)
Comment (ru-fu) [reply doc/versions.json:26]: Resolved but not fixed?

This is needed here and in nrf/doc/nrfxlib/conf.py

File: doc/nrf/known_issues.rst (hunk 7 above)
Comment (greg-fer) [reply doc/nrf/known_issues.rst]: I'll leave this consistency update for a separate task, as there are some more occurrences outside of known issues: https://projecttools.nordicsemi.no/jira/browse/NCSDK-7785

**Review by ru-fu — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Review by nordic-hani — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Review by jdstapes — APPROVED** [review]

**Pull request merged**